cmd/query/app/static_handler.go

```diff
@@ -210,30 +212,49 @@ func loadIndexHTML(open func(string) (http.File, error)) ([]byte, error) {
 	return indexBytes, nil
 }
 
-func loadUIConfig(uiConfig string) (map[string]interface{}, error) {
+func loadUIConfig(uiConfig string) (*UIConfigOptions, error) {
 	if uiConfig == "" {
 		return nil, nil
 	}
 	ext := filepath.Ext(uiConfig)
-	bytes, err := ioutil.ReadFile(filepath.Clean(uiConfig))
+	bytesConfig, err := ioutil.ReadFile(filepath.Clean(uiConfig))
 	if err != nil {
 		return nil, fmt.Errorf("cannot read UI config file %v: %w", uiConfig, err)
 	}
 
 	var c map[string]interface{}
+	var r []byte
+	var re *regexp.Regexp
 	var unmarshal func([]byte, interface{}) error
 
 	switch strings.ToLower(ext) {
 	case ".json":
 		unmarshal = json.Unmarshal
+		re = configPattern
+	case ".js":
+		r = bytes.TrimSpace(bytesConfig)
+		re = configJsPattern
+		if !bytes.Contains(r, []byte("function UIConfig(){")) {
+			return nil, fmt.Errorf("wrong JS function format in UI config file format %v", uiConfig)
+		}
 	default:
 		return nil, fmt.Errorf("unrecognized UI config file format %v", uiConfig)
 	}
 
-	if err := unmarshal(bytes, &c); err != nil {
-		return nil, fmt.Errorf("cannot parse UI config file %v: %w", uiConfig, err)
+	if unmarshal != nil {
+		if err := unmarshal(bytesConfig, &c); err != nil {
+			return nil, fmt.Errorf("cannot parse UI config file %v: %w", uiConfig, err)
+		}
+		// TODO if we want to support other config formats like YAML, we need to normalize `config` to be
+		// suitable for json.Marshal(). For example, YAML parser may return a map that has keys of type
+		// interface{}, and json.Marshal() is unable to serialize it.
```

Comment:
I suggest removing this. We don't have any plans to support anything other than JSON and JS. I would move the unmarshal code directly under `case ".json":` to simplify the logic.

Reply:
can also return directly from switch branches instead of maintaining global-like `r`, `re`, `c` variables.

Reply:
👍